### ISSUE TITLE
python usage example 

### ISSUE THREAD

**Issue by remic33** [facebookresearch/demucs#486]
## ❓ Is there any good python example of demucs usage? 

Hey everyone, 
Trying to use demucs to build a cleaning pipe for audio meeting recording. 
I do not find any clear pythonic usage of demucs, starting with a sample .wav file to seperate audio file into multiple file, one with only voices. 
Am I missing something or it is not in the documentation right now?

**Comment by CarlGao4** [comment]
The APIs are still under development 
You can check it at [CarlGao4/demucs](https://github.com/CarlGao4/demucs), and the docs are under docs/api.md
To install, run `pip install git+https://github.com/CarlGao4/demucs`

**Comment by davies-w** [comment]
Hi Carl,

 I'm looking at your repo API.md, and it looks like you may have implemented what I needed - which is a way to have it separate a pre-existing waveform NP array into other arrays?  Am I reading it correctly? If so would you have an example of how to do it? Below is how I'm loading my MP3 into Jupyter notebooks.

```
from pydub import AudioSegment
audiosegment = AudioSegment.from_mp3("filename.mp3")
wav, sr = np.array(audiosegment.get_array_of_samples()).reshape((-1, audiosegment.channels)), audiosegment.frame_rate
IPython.display.display(IPython.display.Audio(wav, rate=sr))
```

Thanks in advance,
 Winton

**Comment by davies-w** [comment]
Ok, I got it working, apologies for Jupyter-ness:
Kudos Carl!!

```
#@title Load files from GDrive Storage
from google.colab import drive

# Files stored on gdrive.
drive.mount('/content/gdrive', readonly=True) 

# change this line if you have a different song loaded into your GDrive.
!cp gdrive/MyDrive/born.mp3 .

!pip install -q 'git+https://github.com/CarlGao4/demucs'

!pip install pydub

from pydub import AudioSegment
import IPython
import numpy as np
import demucs.api
import pathlib
import torch

audiosegment = AudioSegment.from_mp3("born.mp3")
arr = audiosegment.get_array_of_samples()
sr = audiosegment.frame_rate
ch = audiosegment.channels
aac = np.array(arr, dtype=np.float32).reshape((-1, ch))
aac = aac[30*sr:40*sr,:]
IPython.display.Audio(aac[:, 0], rate=sr)

separator = demucs.api.Separator()
model = separator.load_model()
input = torch.from_numpy(aac.reshape((ch, -1)))
separator.add_track("born", input)
separated = separator.separate_loaded_audio()
for (file, stems) in separated:
    for stem, waveform in stems.items():
      print(stem)
      print(waveform.shape)
      aac_again = waveform.numpy().reshape((-1, ch))
      IPython.display.display(IPython.display.Audio(aac_again[:,0], rate=sr))
```

**Comment by davies-w** [comment]
I do have a remaining question though, I couldn't figure out how to change the default pretrained model, the example you give in the API throws an error.

separator.load_model(name="htdemucs_ft", repo=pathlib.Path("./pretrained"))

**Comment by CarlGao4** [comment]
> If you'd like to use remote models, remove the `repo` parameter

I’d added this line to the docs, thanks for your question!

**Comment by davies-w** [comment]
Thanks Carl - also the name param is model not name.

I've got it working now, but the quality of separation is not at all like the separation that regular demucs gets on the same file, not at all close.

I'm wondering if there's some loss of information about the sample rate or something else? Do you think it should be normalized (1.0,-1.0?)

Thanks again, amazing work!
 W

**Comment by CarlGao4** [comment]
> I'm wondering if there's some loss of information about the sample rate or something else?

I assume that the audio has been resampled to `separator.samplerate` and has the exact amount of channels when calling `add_track`.

> Do you think it should be normalized (1.0,-1.0?)

When?

**Comment by davies-w** [comment]
I don't know, I'm just trying to understand why the separator.add_track doesn't produce the same clean separation as demucs applied to an MP3 file.

**Comment by CarlGao4** [comment]
Then I don’t know why getting this result. Demucs only read the audio and then separate it and save it, without doing anything. Maybe you can try turning off `shifts` and save as wav and see whether you can get the same result?

**Comment by davies-w** [comment]
Also, just to confirm, the audiosegment samplerate is 44.1 khz, with 16 bit integer (I convert to tf.float) - whats interesting is that the MP3 has a bit rate rate of 320K - which doesn't correspond to any combination of those.

I'm not sure what shifts does? I'll try and see though.

W

**Comment by CarlGao4** [comment]
Shifts will randomly select a series of offsets between 0 and 0.5
This can increase the quality sometimes but maybe not all the times
